### PR TITLE
feat(LVGL port): Support for LVGL9 and support for MIPI-DSI screen.

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

# Change description
- Added LVGL9 support
- Added MIPI-DSI support
- Divided into files per feature

### CODE REVIEW

**Comment by espzav** [comment]
@tore-espressif PTAL

File: components/esp_lvgl_port/examples/i2c_oled/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cmake_minimum_required(VERSION 3.16)
```
Comment (tore-espressif) [inline components/esp_lvgl_port/examples/i2c_oled/CMakeLists.txt:1]: I checked the CI and it seems that our `esp_lvgl_port/examples` applications are not built.

Could you please add a CI step to build them? Ideally using idf-build-apps

File: components/esp_lvgl_port/include/esp_lvgl_port_knob.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+#pragma once
+
+#include "esp_err.h"
+#include "lvgl.h"
+
+#if __has_include ("iot_knob.h")
+#include "iot_knob.h"
+#include "iot_button.h"
```
Comment (tore-espressif) [inline components/esp_lvgl_port/include/esp_lvgl_port_knob.h:19]: When I tried adding `knob` without `button` component the compilation failed here.

Knob does not depend on button in the original component. Could we remove the dependency here? (Or at least not fail in this case?)

**Review by tore-espressif — COMMENTED** [review]
A few nitpicks :)

File: components/esp_lvgl_port/include/esp_lvgl_port_knob.h (hunk 2 above)
Comment (espzav) [reply components/esp_lvgl_port/include/esp_lvgl_port_knob.h:19]: We need `button` component for this file: `esp_lvgl_port_knob.c`. When we add knob, there is code for handle click by `button` component.

Comment (tore-espressif) [reply components/esp_lvgl_port/include/esp_lvgl_port_knob.h:19]: I see, could we do something like
``` c
#if __has_include ("iot_knob.h")
#if !__has_include("iot_button.h")
#error LVLG Knob requires button component. Please add it with idf.py add-dependency espressif/button
#endif
#include "iot_knob.h"
#include "iot_button.h"

**Comment by tore-espressif** [comment]
@espzav CI changes LGTM! thank you!

**Comment by espzav** [comment]
@tore-espressif Thank you for review. Is it ok now? Can we merge it?

**Comment by tore-espressif** [comment]
> @tore-espressif Thank you for review. Is it ok now? Can we merge it?
yes, go ahead!

**Pull request merged**